### ISSUE TITLE
Implement logic for the users search bar

### ISSUE THREAD

**Issue by omarelkashef** [uchicago-cs/chigame#260]
The games team created a search bar  #93 where you can choose which model you are searching. Currently, there is a placeholder for User but we need to implement the logic for that. The logic needs to allow for searching by username, first name, last name, or email. If there are more than one matching result, they all should be displayed. If none are matching, the landing page should display such.

**Comment by omarelkashef** [comment]
Looking at what the games team has accomplished, it will be a good idea to keep things consistent and use the same landing page for search results but tweak it for our displaying users instead of games. We will write our code in a way that allows adding a searching functionality in an easy and consistent way.

**Comment by omarelkashef** [comment]
The searching logic we are implementing is appropriate for the "relatively" small number of users we expect to be in the database. If this number grows too much, this logic will need to be imporved.

**Comment by omarelkashef** [comment]
Closing Statement: We added the user searching logic for the search box that takes username or email. Currently, the firstname and lastname fields are not queries since they are None in models.py  We used the same search results landing page that was created by the games team to keep things consistent. We also handled empty search and zero matching search queries. 
@omarelkashef  worked on the query logic for searching
@24raniwalar  worked on edge cases for empty searches and queries with no matching results
